### PR TITLE
[breadcrumb] custom prefix + icons + docs

### PR DESCRIPTION
**Pull request by ericdallo**
* Fix a hardcoded extension :sweat_smile: 
* Add support for icons from `treemacs` if available
* Add `lsp-headerline-breadcrumb-prefix` suggested [here](https://github.com/emacs-lsp/lsp-mode/pull/1899#issuecomment-657265916) and change breadcrumb documentation related:

image

### CODE REVIEW

**Comment by yyoncho** [comment]
1. Can you remove the first space? 
2. Can you use different faces file path and for the symbol path?

**Comment by ericdallo** [comment]
Done @yyoncho!

File: lsp-mode.el [hunk 1]
```diff
@@ -2123,46 +2163,87 @@ The `:global' workspace is global one.")
...
+                ""
+              " ")
+            (lsp--headerline-breadcrumb-arrow-icon)
+            (propertize (f-filename (lsp-workspace-root)) 'font-lock-face lsp-headerline-breadcrumb-project-prefix-face))))
+
+(defun lsp--headerline-breadcrumb-build-file-string ()
+  "Build the file segment string for the breadcrumb."
+  (when (member 'file lsp-headerline-breadcrumb-segments)
+    (format "%s%s %s"
+            (if (= 0 (cl-position 'file lsp-headerline-breadcrumb-segments))
```
Comment (yyoncho) [inline lsp-mode.el:2188]: nit: those should be `(eq (cl-first segments) 'file)`

**Comment by ericdallo** [comment]
Updated Docs: 
image

**Pull request merged**